runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3585,7 +3678,7 @@ remoteCompile(J9VMThread *vmThread, TR::Compilation *compiler, TR_ResolvedMethod
                }
             }
 
-         if (!compiler->getOption(TR_DisableCHOpts) && !useAotCompilation && !compiler->isDeserializedAOTMethod())
+         if (!compiler->getOption(TR_DisableCHOpts) && !useAotCompilation && (compiler->isDeserializedAOTMethodStore() || !compiler->isDeserializedAOTMethod()))
```

Comment:
I'm having trouble understanding this condition. Shouldn't `isDeserializedAOTMethodStore()` imply that this is an AOT compilation?